src/yz_entropy_mgr.erl

```diff
@@ -515,6 +517,35 @@ maybe_poke_tree(S) ->
     end.
 
 %%%===================================================================
+%%% Throttling
+%%%===================================================================
+
+throttle() ->
+    riak_core_throttle:throttle(?YZ_ENTROPY_THROTTLE_KEY).
+
+init_throttle(InitialThrottle) ->
+    riak_core_throttle:init(?YZ_ENTROPY_THROTTLE_KEY,
+                            ?YZ_APP_NAME,
+                            {?YZ_ENTROPY_THROTTLE_LIMITS_KEY,
+                             ?YZ_ENTROPY_THROTTLE_DEFAULT_LIMITS},
+                            {?YZ_ENTROPY_THROTTLE_KILL_KEY, false}),
+    ok = riak_core_throttle:set_throttle(?YZ_ENTROPY_THROTTLE_KEY,
+                                         InitialThrottle).
+
+update_throttle(State) ->
+    case riak_core_throttle:is_throttle_enabled(?YZ_ENTROPY_THROTTLE_KEY) of
+        true ->
+            Load = calculate_current_load(State),
+            riak_core_throttle:set_throttle_by_load(?YZ_ENTROPY_THROTTLE_KEY,
+                                                    Load);
+        false ->
+            ok
+    end.
+
+calculate_current_load(_State) ->
+    yz_solrq:queue_total_length().
```

Comment:
As discussed, let's see if we can use the stat here rather than calling queue_total_length, which will end up doing a gen_server:call to each solrq. See`riak_kv_entropy_manager:get_max_local_vnodeq` for an example. Additionally, and outside this PR, we should probably make queue_total_length use riak_core_util:pmap or something similar so we don't have to wait for each q_worker to respond before calling the next one.